.github/workflows/pypi-package.yml

```diff
@@ -0,0 +1,67 @@
+---
+name: Build & maybe upload PyPI package
+
+on:
+  push:
+    branches: [main]
+    tags: ["*"]
+  pull_request:
+    branches: [main]
+  release:
+    types:
+      - published
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  # Always build & lint package.
+  build-package:
+    name: Build & verify package
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - uses: hynek/build-and-inspect-python-package@v1
+
+  # Upload to Test PyPI on every commit on main.
+  release-test-pypi:
+    name: Publish in-dev package to test.pypi.org
+    if: github.repository_owner == 'python' && github.event_name == 'push' && github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    needs: build-package
+
+    steps:
+      - name: Download packages built by build-and-inspect-python-package
+        uses: actions/download-artifact@v3
+        with:
+          name: Packages
+          path: dist
+
+      - name: Upload package to Test PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          repository-url: https://test.pypi.org/legacy/
+
```

Comment:
Twine only checks a subset of things that PyPI does: I've seen packages pass `twine check` yet fail to upload.

The package is small, we don't need to worry about filling up TestPyPI. I doubt we'll ever get near the limit, and we can safety delete old releases.

And when setting up automation, it sometimes takes a few rounds of iteration to get working: it would be tedious to have to iterate through making many releases.

Reply:
Removed, see: https://github.com/python/python-docs-theme/pull/148#issuecomment-1686473885